tests/test_graph.py

```diff
@@ -42,6 +98,17 @@ def test_bootstrap_dependency_addition():
     assert (('hostpython2' in build_order) or ('hostpython3' in build_order))
 
 
+def test_graph_deplist_transformation():
+    test_pairs = [
```

Comment:
Minor: I'm actually not a big fan of `pytest.mark.parametrize` (and `pytest` in general), but since you started using it above, why didn't you continue?

Reply:
@AndreMiras I don't like it either, and all fully new code I added doesn't use it. I just reused it where the test case was based on a copy&paste of another one :smile:

Reply:
it makes sense that you used what was already in place, I would probably have done the same actualy